cardano-cli/test/cardano-cli-test/Test/Cli/CheckNodeConfiguration.hs

```diff
@@ -0,0 +1,150 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Test.Cli.CheckNodeConfiguration where
+
+import           Cardano.Api
+
+import qualified Cardano.CLI.Run.Debug.CheckNodeConfiguration as CNC
+
+import           Control.Monad (forM_)
+import qualified Data.Aeson as Aeson
+import qualified Data.Aeson.Encode.Pretty as Aeson
+import qualified Data.Aeson.Key as Aeson
+import qualified Data.Aeson.KeyMap as Aeson
+import qualified Data.ByteString.Lazy as LBS
+import           Data.List (isInfixOf)
+import qualified Data.Text as Text
+import qualified Data.Yaml as Yaml
+import           GHC.IO.Exception (ExitCode (..))
+import           System.FilePath ((</>))
+
+import           Test.Cardano.CLI.Util (FileSem, bracketSem, execCardanoCLI, execDetailCardanoCLI,
+                   newFileSem)
+
+import           Hedgehog (Property)
+import qualified Hedgehog as H
+import           Hedgehog.Extras (propertyOnce)
+import qualified Hedgehog.Extras as H
+
+-- | Semaphore protecting against locked file error, when running properties concurrently.
+nodeConfigSem :: FileSem
+nodeConfigSem = newFileSem "test/cardano-cli-test/files/input/check-node-configuration/node-config.json"
+{-# NOINLINE nodeConfigSem #-}
+
+-- Execute this test with:
+-- @cabal test cardano-cli-test --test-options '-p "/check node configuration success/"'@
+hprop_check_node_configuration_success :: Property
+hprop_check_node_configuration_success =
+  propertyOnce $ do
+    -- We test that the command doesn't crash, because otherwise
+    -- execCardanoCLI would fail.
+    bracketSem nodeConfigSem $ \nodeConfigFile -> do
+      H.noteM_ $
+        execCardanoCLI
+          [ "debug"
+          , "check-node-configuration"
+          , "--node-configuration-file"
+          , nodeConfigFile
+          ]
+
+data FiddleKind
+  = FiddleByron
+  | FiddleNonByron
+  deriving (Bounded, Enum)
+
+-- Execute this test with:
+-- @cabal test cardano-cli-test --test-options '-p "/check node configuration fiddle/"'@
+hprop_check_node_configuration_fiddle :: Property
+hprop_check_node_configuration_fiddle = do
+  let supplyValues :: [(FilePath, FiddleKind)] =
+        [ (path, fiddleKind)
+        | path <-
+            [ "test/cardano-cli-test/files/input/check-node-configuration/node-config.json"
+            , "test/cardano-cli-test/files/input/check-node-configuration/node-config.yaml"
+            ]
+        , fiddleKind <- [minBound .. maxBound]
+        ]
+
+  propertyOnce $ forM_ supplyValues $ \(nodeConfigPath, fiddleKind) -> H.moduleWorkspace "tmp" $ \tempDir ->
+    bracketSem nodeConfigSem $ \_ -> do
+      let finalInputConfig = tempDir </> "node-config-changed.json"
+          -- TODO why is that writing to "AlonzoGenesisHash" writes one more 0 than specified here? (and hence the need for this hack)
+          wrongHash = Text.pack $ replicate (case fiddleKind of FiddleByron -> 65; FiddleNonByron -> 64) '0'
```

Comment:
This needs to be sorted out before merging. This one is actually quite surprising.